portal/index.php

```diff
@@ -73,36 +74,63 @@
  * Also verified as the portal account id is rebuilt from patient data
  * and compared to portal credential account id lookup.
  * */
-if (!empty($_GET['service_auth'] ?? null)) {
-    $token = $_GET['service_auth'];
-    $redirect_token = $_GET['target'] ?? null;
-    try {
-        $oneTime = new OneTimeAuth();
-        $auth = $oneTime->processOnetime($token, $redirect_token);
-        $logit->portalLog('onetime login attempt', $auth['pid'], 'patient logged in and redirecting', '', '1');
-        exit();
-    } catch (OneTimeAuthExpiredException $exception) {
-        $logit->portalLog(
-            'onetime login attempt',
-            $exception->getPid() ?? '',
-            ':invalid one time',
-            '',
-            '0'
-        );
-        OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
-        header('Location: ' . $landingpage . '&oe');
-        exit();
-    } catch (OneTimeAuthException $exception) {
-        $logit->portalLog(
-            'onetime login attempt',
-            $exception->getPid() ?? '',
-            ':invalid one time',
-            '',
-            '0'
-        );
-        OpenEMR\Common\Session\SessionUtil::portalSessionCookieDestroy();
-        header('Location: ' . $landingpage . '&oi');
-        exit();
+if (!empty($_REQUEST['service_auth'] ?? null)) {
+
+    if (!empty($_GET['service_auth'] ?? null)) {
+        // we have to setup the csrf key to preven CSRF Login attacks
+        // we also implement this mechanism in order to handle Same-Site cookie blocking when being referred by
+        // an external site domain.  We used to auto process via GET but now we submit via the POST in order to make it
+        // a same site cookie origin request. This is a workaround for the Same-Site cookie blocking.
+        CsrfUtils::setupCsrfKey();
+        $twig = new TwigContainer(null, $GLOBALS['kernel']);
+        echo $twig->getTwig()->render('portal/login/autologin.html.twig', [
+            'action' => $GLOBALS['web_root'] . '/portal/index.php',
+            'service_auth' => $_GET['service_auth'],
+            'target' => $_GET['target'] ?? null,
+            'csrf_token' => CsrfUtils::collectCsrfToken('autologin'),
+            'pagetitle' => xl("OpenEMR Patient Portal"),
+            'images_static_relative' => $GLOBALS['images_static_relative'] ?? ''
+        ]);
+        exit;
+    }
+    else if (!empty($_POST['service_auth'] ?? null)) {
```

Comment:
Isn't this always true and valid because any POST will be in REQUEST!

Reply:
It gets caught in the first if statement with the $_GET request.

Reply:
oh shoot sorry, I missed this is in inner conditional. Don't pay attention to the man behind the curtain!:)

Reply:
I don't know... the man behind the curtain ultimately gets you back to Kansas...